source/patterns/@aws-solutions-constructs/core/lib/bedrock-inference-helper.ts

```diff
@@ -0,0 +1,140 @@
+/**
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ *  with the License. A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
+ *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ */
+
+// Imports
+import * as cdk from 'aws-cdk-lib';
+import * as bedrock from 'aws-cdk-lib/aws-bedrock';
+import * as defaults from '../index';
+// Note: To ensure CDKv2 compatibility, keep the import statement for Construct separate
+import { Construct } from 'constructs';
+
+export interface BuildInferenceProfileProps {
+  readonly bedrockModelId: string;
+  readonly inferenceProfileProps?: bedrock.CfnApplicationInferenceProfileProps;
+  readonly deployCrossRegionProfile?: boolean;
+}
+
+export interface BuildInferenceProfileReponse {
+  readonly crossRegion?: boolean,
+  readonly inferenceProfile: bedrock.CfnApplicationInferenceProfile,
+}
+
+export function buildInferenceProfile(scope: Construct, id: string, props: BuildInferenceProfileProps): BuildInferenceProfileReponse {
+  const areaMap = createAreaPrefixMapping(scope, id);
+
+  const regionPrefix = cdk.Fn.select(0, cdk.Fn.split('-', cdk.Aws.REGION));
+  const crossRegion = IsCrossRegionProfile(props.deployCrossRegionProfile);
+  const inferenceSourceArn = crossRegion ?
+    `arn:${cdk.Aws.PARTITION}:bedrock:${cdk.Aws.REGION}:${cdk.Aws.ACCOUNT_ID}:inference-profile/${areaMap.mapping.findInMap(regionPrefix, "prefix")}.${props.bedrockModelId}` :
+    `arn:${cdk.Aws.PARTITION}:bedrock:${cdk.Aws.REGION}:${cdk.Aws.ACCOUNT_ID}:foundation-model/${props.bedrockModelId}`;
+
+  const constructInferenceProps: bedrock.CfnApplicationInferenceProfileProps = {
+    modelSource: {
+      copyFrom: inferenceSourceArn
+    },
+    inferenceProfileName: defaults.generatePhysicalInferenceProfileName("", [id])
+  };
+  const finalProps = defaults.consolidateProps(constructInferenceProps, props.inferenceProfileProps);
+  const inferenceProfile = new bedrock.CfnApplicationInferenceProfile(scope, `${id}-inference`, finalProps);
+
+  return {
+    inferenceProfile,
+    crossRegion
+  };
+}
+
+export interface MappingResponse {
+  readonly mappingName: string,
+  readonly mapping: cdk.CfnMapping,
+}
+
+export function createAreaPrefixMapping(scope: Construct, id: string): MappingResponse {
+  const mappingName = defaults.removeNonAlphanumeric(`${id}-area-prefix-mapping`);
+  const newMapping = new cdk.CfnMapping(scope, `${id}-area-prefix-mapping`, {
+    mapping: {
```

Comment:
no `af` or `sa` in this list. Is it because they dont have inference profiles. I think bedrock models are available at least. May be this mapping is only required for regions with cross-region inference profile it seems